usr/src/uts/common/fs/zfs/spa_history.c

```diff
@@ -455,6 +455,7 @@ log_internal(nvlist_t *nvl, const char *operation, spa_t *spa,
 	fnvlist_add_string(nvl, ZPOOL_HIST_INT_NAME, operation);
 	fnvlist_add_uint64(nvl, ZPOOL_HIST_TXG, tx->tx_txg);
 
+	zfs_ereport_spa_history(spa, nvl);
```

Comment:
Let me answer several questions at once:

- @gdamore FreeBSD solves the slow consumer problem by using a single userspace daemon that communicates with the kernel: devd(8).  devd buffers sysevents for other consumers, and also executes shell commands.  Slow consumers may miss events, but they won't cause any other consumers to miss events.

- @gdamore this PR doesn't change any existing events; it only adds new ones.  The change from "create" to "pool create" is only changing a line in the zpool history file.  It's not yet exposed through any sysevent.

- @gdamore @richardelling This change doesn't cause the "zpool history" file to grow; it just emits sysevents whenever history is logged.  In my experience, snapshot creation is limited to less than ten per second.  Is that too many for Illumos?  If so, could Illumos use a sysctl to gate the emission of those events?

- @gdamore I don't like the idea of a long-poll ioctl.  It sounds good, but I think it would lead to problems like Linux's `io_getevents`.  `io_getevents` sounded like a good idea at the time, but a lot of programs want a single blocking system call that will return whenever any aio request is complete _or_ a socket is ready.  Instead, they must use two system calls (`io_getevents` and `epoll`) in two separate threads or else do gymnastics with `signalfd`.  In our case a ZFS monitoring daemon wants to be notified when ZFS's state changes _or_ some other sysevent arrives, such as a device arrival notification.  As long as ZFS's state change notifications use sysevents; it's easy.  But if ZFS state change notifications are implemented with a long-poll ioctl, then the monitoring daemon will need two threads.

- @yuripv It looks like Nexenta took the approach of adding sysevents for each and every interesting event, as I suggested in my first reply to @ahrens.  IMHO, that takes too much repetitive code

- @bahamas10 #101 looks like it's almost exactly the same as this PR.  Naturally, I like it ;).  My one complaint is that #101's sysevents contain hierarchical nvlists.  Would it be possible to flatten them out?